src/createPackage.ts

```diff
@@ -1,74 +1,47 @@
-  // list all of the solidity interfaces
-  glob(interfacesGlob, (err, interfacePaths) => {
-    if (err) throw err;
-
-    // for each interface path
-    for (const interfacePath of interfacePaths) {
-      const interfaceFile = fse.readFileSync(interfacePath, 'utf8');
-      const relativeInterfaceFile = transformRemappings(interfaceFile);
-
-      const contractPath = interfacePath.substring(interfacesDir.length + 1);
-      fse.outputFileSync(path.join(interfacesDestination, contractPath), relativeInterfaceFile);
-
-      // get the interface name
-      interfaceName = interfacePath.substring(interfacePath.lastIndexOf('/') + 1, interfacePath.lastIndexOf('.'));
-
-      // copy interface abi to the export directory
-      fse.copySync(`${outDir}/${interfaceName}.sol/${interfaceName}.json`, `${abiDestination}/${interfaceName}.json`);
-    }
-    console.log(`Copied ${interfacePaths.length} interfaces`);
-
-    if (typingType === TypingType.CONTRACTS) {
-      glob(contractsGlob, (err, contractPaths) => {
-        if (err) throw err;
-
-        for (const contractPath of contractPaths) {
-          const contractFile = fse.readFileSync(contractPath, 'utf8');
-          const relativeContractFile = transformRemappings(contractFile);
-
-          const relativeContractPath = contractPath.substring(contractsExportDir.length + 1);
-          fse.outputFileSync(path.join(contractsDestination, relativeContractPath), relativeContractFile);
-
-          const contractName = contractPath.substring(contractPath.lastIndexOf('/') + 1, contractPath.lastIndexOf('.'));
-          fse.copySync(`${outDir}/${contractName}.sol/${contractName}.json`, `${abiDestination}/${contractName}.json`);
-        }
-        console.log(`Copied ${contractPaths.length} contracts`);
-      });
-    }
-
-    createReadmeAndLicense(packageJson.name, typingType, exportDir, interfaceName);
-
-    // install package dependencies
-    console.log(`Installing abi dependencies`);
-    execSync(`cd ${exportDir} && yarn`);
-  });
+  // Empty export destination directory
+  const destinationDir = `export/${packageName}-${exportType}`;
+  fse.emptyDirSync(destinationDir);
+
+  console.log('Installing dependencies');
+  execSync('yarn');
+
+  // Read and copy the input package.json
+  const inputPackageJson = fse.readJsonSync('./package.json');
+  if (!inputPackageJson) throw new Error('package.json not found');
+  // Create custom package.json in the export directory
+  const packageJson: PackageJson = {
+    name: packageName,
+    version: inputPackageJson.version,
+    dependencies: {
+      ...inputPackageJson.dependencies,
+    },
+  };
+  fse.writeJsonSync(`${destinationDir}/package.json`, packageJson, { spaces: 4 });
+
+  // Copy the interfaces and their ABIs
+  copySolidityFiles(outDir, interfacesDir, destinationDir);
+
+  // Copy the contracts only if the export type is contracts
+  if (exportType === ExportType.CONTRACTS) copySolidityFiles(outDir, contractsDir, destinationDir);
```

Comment:
Idea: Instead of exporting the interfaces ABIs as default, could we have an if for each `exportType`?
If the user wants to only export the interfaces he can set `interfaces` as argument. If he wants to export only the contracts he can set `contracts`. If he wants both, he could set `all`.
By this way, we wouldn't be forcing to export the interfaces if the user only wants to export contracts

Reply:
👍🏻 added this to Linear to think about later.